mm/slub.c

```diff
@@ -3925,11 +3944,7 @@ void *__kmalloc(size_t size, gfp_t flags)
 	if (unlikely(ZERO_OR_NULL_PTR(s)))
 		return s;
 
-	ret = kfence_alloc_with_size(s, size, flags);
-	if (ret)
-		return ret;
-
-	ret = slab_alloc(s, flags, _RET_IP_);
+	ret = slab_alloc(s, flags, RET_IP_SIZE(size));
```

Comment:
I think this will never pass upstream code review.

Both slab_alloc_node and slab_alloc are __always_inline. So any additional parameter, if unused, will be optimized out.

So, I think just adding another argument with size and then just passing 'size' instead of 'addr' to KFENCE is better. It also removes the bit-twiddling.

Reply:
Hm, you are right, `_RET_IP_` is only passed to __slab_alloc, otherwise it just dissolves.
Adding another argument sure looks cleaner.